Board Logic

SteveNilla opened the pull request:
Board
* created board object with room / hallway identifiers
* established basic player move logic

BoardRoom
* created room object with name and player occupancy attributes

BoardHallway
* created hallway extension of room

MoveRequest
* modified MoveRequest to enumerate specific move set

Player
* created basic getter/setter for player board position

SteveNilla committed:
Board Logic

Board
* created board object with room / hallway identifiers
* established basic player move logic

BoardRoom
* created room object with name and player occupancy attributes

BoardHallway
* created hallway extension of room

MoveRequest
* modified MoveRequest to enumerate specific move set

Player
* created basic getter/setter for player board position

Kit-Kindred commented on Source_Code/Common/Messages/ActionRequests/MoveRequest.java:
```diff
@@ -3,21 +3,34 @@
-   public int PlayerID;
-   
-   public MoveRequest( int PlayerID, String moveDirection )
+   enum Move
```
This needs to be public to support our current ClueLessClient main (since we have MoveRequest sends in master)

Kit-Kindred commented on Source_Code/Common/Messages/ActionRequests/MoveRequest.java:
```diff
@@ -3,21 +3,34 @@
+    *
+    * @param args
+    */
+   public static void main(String[] args) 
```
Do we want to comment this out since it's more an example than anything else?

Kit-Kindred commented on Source_Code/Common/Board.java:
```diff
@@ -0,0 +1,150 @@
+    int currY = player.getyPos();
+
+    // new player x,y positions
+    int newX;
+    int newY;
```
Let's initialize these to currX and currY or add a default to the switch case. Otherwise my compiler freaks out about potential non-initialization

Kit-Kindred commented on Source_Code/Common/Board.java:
```diff
@@ -0,0 +1,150 @@
+    int newX;
+    int newY;
+
+    // set prospective player location
```
I'm not sure how up-tight my compiler settings are, but it doesn't like this switch. Had to make it switch(move.moveDirection) and all of the options "case UP", "case DOWN", etc. to get it to compile

Kit-Kindred commented on Source_Code/Common/Board.java:
```diff
@@ -0,0 +1,150 @@
+    // if room is an occupied hallway, player cannot move there
+    if (board[currX][currY] instanceof BoardHallway)
+    {
+      if (board[currX][currY].isOccupied())
```
Again with up-tight settings. It won't allow implicit casting, so I had to change this to "if (((BoardHallway)board[currX][currY]).isOccupied())"

Kit-Kindred reviewed (changes requested):
Steve, I think the architecture and mechanization is good. There are a couple of changes needed to make my compiler happy though.

dav96che committed:
Added the PlayerHand and SolutionHand classes.
Created the player and solution hands to keep track of cards as they are
used in the game.

dav96che committed:
Modified PlayerHand to use an ArrayList for each card and added
CardDeck.

Players can have more than one of each card, so the cards for each type
will be stored in an array list (a regular array would also work).
Added the CardDeck class, which handles shuffling the cards and
assigning each player their hands.

dav96che committed:
Removed the shuffleAndAssignCards method from the server class.

dav96che committed:
Fixed shuffling mechanic to use the entire deck.

Added some better string outputs to each card class as well as the hand
classes. The Solution hand takes the top three cards of each type, and
the rest of the cards get shuffled and assigned to players.

dav96che committed:
Added the PlayerHandUpdate message.

THe PlayerHandUpdate will send the hand instance to the user play on the
client. Since we're not using serialization currently, the message
carries a string that will be interpreted on the client.

dav96che committed:
Update CardDeck.java

Updated based on Kit's comments on this class.

Kit-Kindred committed:
Merge pull request #56 from Kit-Kindred/Player_Card_Handling

Player card handling
